executing/executing.py

```diff
@@ -446,40 +483,52 @@ class Executing(object):
     """
 
     def __init__(self, frame, source, node, stmts, decorator):
+        # type: (types.FrameType, Source, ast.AST, Sequence[ast.stmt], Any) -> None
         self.frame = frame
         self.source = source
         self.node = node
         self.statements = stmts
         self.decorator = decorator
 
     def code_qualname(self):
+        # type: () -> str
         return self.source.code_qualname(self.frame.f_code)
 
     def text(self):
+        # type: () -> str
         return self.source.asttokens().get_text(self.node)
 
     def text_range(self):
+        # type: () -> Tuple[int, int]
         return self.source.asttokens().get_text_range(self.node)
 
 
 class QualnameVisitor(ast.NodeVisitor):
     def __init__(self):
+        # type: () -> None
         super(QualnameVisitor, self).__init__()
-        self.stack = []
-        self.qualnames = {}
+        self.stack = [] # type: List[str]
+        self.qualnames = {} # type: Dict[Tuple[str, int], str]
 
     def add_qualname(self, node, name=None):
-        name = name or node.name
+        # type: (ast.AST, Optional[str]) -> None
+        name = name or node.name # type: ignore[attr-defined]
         self.stack.append(name)
         if getattr(node, 'decorator_list', ()):
-            lineno = node.decorator_list[0].lineno
+            lineno = node.decorator_list[0].lineno # type: ignore[attr-defined]
         else:
-            lineno = node.lineno
+            lineno = node.lineno # type: ignore[attr-defined]
         self.qualnames.setdefault((name, lineno), ".".join(self.stack))
 
     def visit_FunctionDef(self, node, name=None):
+        # type: (ast.AST, Optional[str]) -> None
+        if sys.version_info[0] == 3:
+            assert isinstance(node, ast.FunctionDef) or isinstance(node, ast.AsyncFunctionDef) or isinstance(node, ast.Lambda), node
+        else:
+            assert isinstance(node, ast.FunctionDef) or isinstance(node, ast.Lambda), node
```

Comment:
Does this work?

```suggestion
        assert isinstance(node, (ast.FunctionDef, ast.Lambda)) or sys.version_info[0] == 3 and isinstance(node, ast.AsyncFunctionDef), node
```

Reply:
Yup, seems to work fine

Reply:
If you can't collapse the `if`, can you at least use `isinstance` with a tuple?